File: leetcode/53.maximum-subarray/go/maxsubarray.go
package maxsubarray

func maxSubArray(nums []int) int {
	if len(nums) == 0 {
		return 0
	}

	res := nums[0]
	cur := 0
	for _, v := range nums {
		cur += v

		if cur > res {
			res = cur
		}

		if cur < 0 {
			cur = 0
		}

	}

	return res
}

File: leetcode/53.maximum-subarray/cpp/maxsubarray.cpp
#include <vector>

class Solution
{
public:
    int maxSubArray(std::vector<int> &nums)
    {
        if (nums.size() == 0)
            return 0;

        int res = nums[0];
        int cur = 0;

        for (auto v : nums)
        {
            cur += v;

            if (cur > res)
                res = cur;

            if (cur < 0)
                cur = 0;
        }

        return res;
    }
};
